src/render.rs

```diff
@@ -167,22 +166,64 @@ impl<'a> Renderer<'a> {
         }
     }
 
-    // TODO: clean up this, too ugly right now for the == and != nodes
-    fn eval_condition(&self, node: Node) -> TeraResult<bool> {
+    fn eval_expression(&self, node: Node) -> TeraResult<Value> {
         match node {
             // Simple truthiness check
             Identifier { .. } => {
                 let value = try!(self.eval_ident(&node));
-                Ok(value.is_truthy())
+                Ok(value)
+            },
+            l @ Logic { .. } => {
+                let value = try!(self.eval_condition(l));
+                Ok(Value::Bool(value))
+            },
+            m @ Math { .. } => {
+                let result = try!(self.eval_math(&m));
+                Ok(Value::F64(result as f64))
+            },
+            Int(val) => {
+                Ok(Value::I64(val as i64))
+            },
+            Float(val) => {
+                Ok(Value::F64(val as f64))
+            },
+            Bool(b) => {
+                Ok(Value::Bool(b))
+            },
+            _ => unreachable!()
+        }
+    }
+
+    // TODO: clean up this, too ugly right now for the == and != nodes
+    fn eval_condition(&self, node: Node) -> TeraResult<bool> {
+        match node {
+            Identifier { .. } => {
+                Ok(self.eval_ident(&node).map(|v| v.is_truthy()).unwrap_or(false))
+            },
+            Test { expression, name, params } => {
+                let tester = try!(self.tera.get_tester(&name));
+                let context = match self.context.as_object() {
+                    Some(map) => map,
+                    None => return Err(Internal("context is not an object.".into()))
+                };
```

Comment:
I think you can safely remove that, the next time the context might change is when moving from JSON to something specific to Tera and those lines will not be relevant then

You can safely unwrap